dask/sizeof.py

```diff
@@ -267,10 +267,18 @@ def register_xarray():
     XARRAY_VERSION = Version(xr.__version__)
     XARRAY_GE_2024_02 = XARRAY_VERSION >= Version("2024.02.0")
 
+    @sizeof.register(xr.core.utils.Frozen)
+    def xarray_sizeof_frozen(obj):
+        return sys.getsizeof(obj) + sizeof(obj.mapping)
+
     @sizeof.register(xr.DataArray)
-    @sizeof.register(xr.Dataset)
+    @sizeof.register(xr.Variable)
     def xarray_sizeof_da(obj):
-        return obj.nbytes
+        return sys.getsizeof(obj) + sizeof(obj.data)
+
+    @sizeof.register(xr.Dataset)
+    def xarray_sizeof_ds(obj):
+        return sys.getsizeof(obj) + sizeof(obj.variables)
```

Comment:
I guess there is stuff in a Dataset that I am not tracking here. @dcherian I would appreciate a review. If we're off by a couple percent that's not too bad but we'd like to capture the bulk of the data if possible